File: objectiveai-sdk-py/objectiveai_sdk/functions/profiles/http.py
"""HTTP functions for function profile endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai_sdk.client import ObjectiveAI
    from objectiveai_sdk.functions.profiles import (
        GetProfileRequest,
        GetProfileResponse,
        ListProfileResponse,
        ListProfilesRequest,
        UsageProfileResponse,
    )


async def list_profiles(
    client: ObjectiveAI, params: ListProfilesRequest,
) -> ListProfileResponse:
    """List all profiles accessible to the authenticated user."""
    return await client.post_unary("functions/profiles/list", params)


async def get_profile(
    client: ObjectiveAI, params: GetProfileRequest,
) -> GetProfileResponse:
    """Retrieve a profile definition."""
    return await client.post_unary("functions/profiles", params)


async def get_profile_usage(
    client: ObjectiveAI, params: GetProfileRequest,
) -> UsageProfileResponse:
    """Retrieve usage statistics for a specific profile."""
    return await client.post_unary("functions/profiles/usage", params)

File: objectiveai-sdk-rs/src/functions/profiles/http.rs
//! HTTP functions for profile management.

use crate::{HttpClient, HttpError};

/// Lists all profiles accessible to the authenticated user.
pub async fn list_profiles(
    client: &HttpClient,
    params: super::request::ListProfilesRequest,
) -> Result<super::response::ListProfileResponse, HttpError> {
    client
        .send_unary(
            reqwest::Method::POST,
            "functions/profiles/list",
            Some(params),
        )
        .await
}

/// Retrieves a profile definition.
pub async fn get_profile(
    client: &HttpClient,
    params: super::request::GetProfileRequest,
) -> Result<super::response::GetProfileResponse, HttpError> {
    client
        .send_unary(reqwest::Method::POST, "functions/profiles", Some(params))
        .await
}

/// Gets usage statistics for a specific profile.
pub async fn get_profile_usage(
    client: &HttpClient,
    params: super::request::GetProfileRequest,
) -> Result<super::response::UsageProfileResponse, HttpError> {
    client
        .send_unary(
            reqwest::Method::POST,
            "functions/profiles/usage",
            Some(params),
        )
        .await
}
